iotileship/iotile/ship/actions/ModifyJsonStep.py

```diff
@@ -48,6 +48,7 @@ class ModifyJsonStep: # pylint: disable=too-few-public-methods
     Args:
         key (list): A list of keys and subkeys specifing the key to change (list can be one)
         value (str): The value to change for the above key
+        value_type (str, optional): The type of value. Typedargs types supported Defaults to 'string'
```

Comment:
I think that, conventionally, the item you put in the parentheses should be referring to the type of the argument. It would be clearer to state that it's an optional argument in the description of the arg.

Also, shouldn't arguments that are optional be grouped at the end (as kwargs)? I know it might not matter in this case since they're all passed under a single "args" input, but in normal functions that is the expected pattern.

Reply:
Only example I could (easily) find for optional arguments in docstrings suggested that. But I rearranged it.

The value_type being optional directly modifies the value argument so I feel like it should be documented right next to it.